commit d6c4de90902a005fc14e55c1789bec276dbfcdf0
Author: Ruslan Kabatsayev <b7.10110111@gmail.com>
Date:   Wed Mar 14 15:04:35 2012 +0400

    Check if GdkWindow is destroyed before restoring composited state
    CCBUG: 295875

diff --git a/src/animations/oxygeninnershadowdata.cpp b/src/animations/oxygeninnershadowdata.cpp
index d3cf05e..06d5072 100644
--- a/src/animations/oxygeninnershadowdata.cpp
+++ b/src/animations/oxygeninnershadowdata.cpp
@@ -164,7 +164,7 @@ namespace Oxygen
 
         // remove compositing flag
         GdkWindow* window( gtk_widget_get_window( widget ) );
-        if( GDK_IS_WINDOW( window ) && G_OBJECT_TYPE_NAME( widget ) != std::string("GtkPizza") )
+        if( GDK_IS_WINDOW( window ) && !gdk_window_is_destroyed(window) && G_OBJECT_TYPE_NAME( widget ) != std::string("GtkPizza") )
         { gdk_window_set_composited( window, _initiallyComposited ); }
     }
 
